spec/Appendix B -- Grammar Summary.md

```diff
@@ -162,7 +162,7 @@ ObjectField[Const] : Name : Value[?Const]
 
 VariableDefinitions : ( VariableDefinition+ )
 
-VariableDefinition : Variable : Type DefaultValue?
+VariableDefinition : Variable : Type Directives? DefaultValue?
```

Comment:
Putting the directives before the default value seems to conform closer to how directives are used in general, but I could be convinced that this should be `VariableDefinition : Variable : Type DefaultValue? Directives?` instead.

The reason this seems to match better, is due to things like `Union` and `Fragment` definitions. 

For a union definition, it looks like:
`union SomeUnion @some_directive = SomeObject | SomeOtherObject`

For a fragment, it's:
`fragment SomeFragment on SomeObject @directives { ... }`

So the directive always goes after the type is defined, but before the definition's "value" is defined. In this case, a default value is analogous to a fragment's FieldSet, or a union's UnionMemberTypes, I think.

Reply:
@mjmahone It should be after default value to match existing syntax for input value definition:
http://facebook.github.io/graphql/June2018/#InputValueDefinition

Reply:
Also please use `Directives[Const]` since you can use variables before you defined them.

Reply:
Wow I missed the InputValueDefinition. That seems like an unfortunate choice we made, as it doesn't feel like the natural "fit" given where directives usually are put (and how rare directives on InputValueDefinitions are likely to be). But given that, I agree, much better to make the change, and keep the two aligned.

And the const-ness is a very good point: you wouldn't want variables inside your directive arguments as it's impossible to define them beforehand. Thank you!